apps/prairielearn/src/lib/db-types.ts

```diff
@@ -46,73 +48,73 @@ export const JsonCommentSchema = z.union([z.string(), z.array(z.any()), z.record
 export const SprocAssessmentsFormatForQuestionSchema = z.array(
   z.object({
-    label: z.string(),
     assessment_id: IdSchema,
+    color: z.string(),
     course_instance_id: IdSchema,
+    label: z.string(),
     share_source_publicly: z.boolean(),
-    color: z.string(),
   }),
 );
 
 // Result of check_assessment_access sproc
 const SprocCheckAssessmentAccessSchema = z.object({
+  active: z.boolean().nullable(),
   credit: z.union([z.string(), z.literal('None')]),
-  time_limit_min: z.union([z.string(), z.literal('—')]),
-  start_date: z.union([z.string(), z.literal('—')]),
   end_date: z.union([z.string(), z.literal('—')]),
   mode: EnumModeSchema.nullable(),
-  active: z.boolean().nullable(),
+  start_date: z.union([z.string(), z.literal('—')]),
+  time_limit_min: z.union([z.string(), z.literal('—')]),
 });
 
 // Result of authz_assessment sproc
 export const SprocAuthzAssessmentSchema = z.object({
+  access_rules: z.array(SprocCheckAssessmentAccessSchema),
+  active: z.boolean(),
   authorized: z.boolean(),
-  exam_access_end: DateFromISOString.nullable(),
   credit: z.number().nullable(),
   credit_date_string: z.string().nullable(),
-  time_limit_min: z.number().nullable(),
-  password: z.string().nullable(),
+  exam_access_end: DateFromISOString.nullable(),
   mode: EnumModeSchema.nullable(),
+  next_active_time: z.string().nullable(),
+  password: z.string().nullable(),
   show_closed_assessment: z.boolean(),
   show_closed_assessment_score: z.boolean(),
-  active: z.boolean(),
-  next_active_time: z.string().nullable(),
-  access_rules: z.array(SprocCheckAssessmentAccessSchema),
+  time_limit_min: z.number().nullable(),
 });
 
 // Result of authz_assessment_instance sproc
 export const SprocAuthzAssessmentInstanceSchema = z.object({
+  access_rules: z.array(SprocCheckAssessmentAccessSchema),
+  active: z.boolean(),
   authorized: z.boolean(),
   authorized_edit: z.boolean(),
-  exam_access_end: DateFromISOString.nullable(),
   credit: z.number().nullable(),
   credit_date_string: z.string().nullable(),
-  time_limit_min: z.number().nullable(),
-  time_limit_expired: z.boolean(),
-  password: z.string().nullable(),
+  exam_access_end: DateFromISOString.nullable(),
   mode: EnumModeSchema.nullable(),
+  next_active_time: z.string().nullable(),
+  password: z.string().nullable(),
   show_closed_assessment: z.boolean(),
   show_closed_assessment_score: z.boolean(),
-  active: z.boolean(),
-  next_active_time: z.string().nullable(),
-  access_rules: z.array(SprocCheckAssessmentAccessSchema),
+  time_limit_expired: z.boolean(),
+  time_limit_min: z.number().nullable(),
 });
 
 // Result of authz_course sproc
 export const SprocAuthzCourseSchema = z.object({
   course_role: z.enum(['None', 'Previewer', 'Viewer', 'Editor', 'Owner']),
-  has_course_permission_own: z.boolean(),
   has_course_permission_edit: z.boolean(),
-  has_course_permission_view: z.boolean(),
+  has_course_permission_own: z.boolean(),
```

Comment:
Same comment as the inline arrays: things were sorted by decreasing permission order. I'm willing to roll with the change though.